crates/ink/codegen/src/generator/metadata.rs

```diff
@@ -442,4 +514,29 @@ mod tests {
             ],
         )
     }
+
+    #[test]
+    fn constructor_return_type_works() {
+        let expected_no_ret_type_spec = ":: ink :: metadata :: ReturnTypeSpec :: new (:: core :: option :: Option :: None)";
```

Comment:
It's not a great way to test it using strings like this. It's brittle and dependent upon the codegen impl. 

If we can come up with a good solution with types + reflection this will be easier to test.